subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -174,13 +223,18 @@ void bl_boot(const struct fw_info *fw_info)
 		"   b   clear\n"
 		"out:\n"
 		"   dsb\n"
+#ifndef CONFIG_SB_DISABLE_SELF_R_X
 		/* Jump to reset vector of an app */
 		"   bx r0\n"
+#endif
 		:
 		: "r" (vector_table[1]), "i" (CONFIG_SRAM_BASE_ADDRESS),
 		  "i" (CONFIG_SRAM_SIZE * 1024), "i" (0)
 		: "r0", "r1", "r2", "r3", "memory"
 	);
+#elif defined(CONFIG_SB_DISABLE_SELF_R_X)
```

Comment:
```suggestion
#if defined(CONFIG_SB_DISABLE_SELF_R_X)
```

This option is not supposed to be an alternative to CONFIG_SB_CLEANUP_RAM.